lib/clone.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+// Simple clone function, which creates a copy of the given node
+// And recurses on the children (due to the shallow nature of the copy)
```

Comment:
think you could be more clear, e.g. "creates a deep copy of the given node" and "due to the shallow nature of the mathjs node clone"